<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1117</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Osmium</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1" alignment="Qt::AlignmentFlag::AlignLeft">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="minimumSize">
       <size>
        <width>225</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>225</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Channel Options</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item alignment="Qt::AlignmentFlag::AlignTop">
        <widget class="QWidget" name="widget" native="true">
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="1" column="0">
           <widget class="QCheckBox" name="chbInheritOpts">
            <property name="statusTip">
             <string>Whether this channel should inherit the below values from the default settings.</string>
            </property>
            <property name="text">
             <string>Use Default</string>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="btnResetOpts">
            <property name="text">
             <string>Reset to Default</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QComboBox" name="cmbChannel"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QScrollArea" name="scraChannelOpts">
         <property name="frameShape">
          <enum>QFrame::Shape::NoFrame</enum>
         </property>
         <property name="verticalScrollBarPolicy">
          <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_3">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>240</width>
            <height>340</height>
           </rect>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <property name="topMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Wave Color</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QWidget" name="widget_4" native="true">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLineEdit" name="lineEdit"/>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>Wave Thickness</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="dsbWaveThickness">
             <property name="minimum">
              <double>0.500000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Scope Width</string>
             </property>
             <property name="buddy">
              <cstring>sbScopeWidth</cstring>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="sbScopeWidth">
             <property name="suffix">
              <string>ms</string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
             <property name="value">
              <number>40</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Amplification</string>
             </property>
             <property name="buddy">
              <cstring>dsbAmplification</cstring>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="dsbAmplification">
             <property name="minimum">
              <double>0.100000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QCheckBox" name="chbStereo">
             <property name="text">
              <string>Stereo</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0" colspan="2">
            <widget class="Line" name="line">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_8">
             <property name="statusTip">
              <string>How far forward the scope can &quot;nudge&quot; waveforms so they line up from frame to frame.</string>
             </property>
             <property name="text">
              <string>Maximum Nudge</string>
             </property>
             <property name="buddy">
              <cstring>sbMaxNudge</cstring>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QSpinBox" name="sbMaxNudge">
             <property name="statusTip">
              <string>How far forward the scope can &quot;nudge&quot; waveforms so they line up from frame to frame.</string>
             </property>
             <property name="suffix">
              <string>ms</string>
             </property>
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>16</number>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_11">
             <property name="statusTip">
              <string>How high (proportional to the largest sample in the window) a wave must get for it to count as a rising edge.</string>
             </property>
             <property name="text">
              <string>Trigger Threshold</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QDoubleSpinBox" name="dsbTriggerThreshold">
             <property name="statusTip">
              <string>How high (proportional to the largest sample in the window) a wave must get for it to count as a rising edge.</string>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.050000000000000</double>
             </property>
             <property name="value">
              <double>0.250000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>225</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>225</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Global Options</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QScrollArea" name="scrollArea">
         <property name="frameShape">
          <enum>QFrame::Shape::NoFrame</enum>
         </property>
         <property name="verticalScrollBarPolicy">
          <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>249</width>
            <height>432</height>
           </rect>
          </property>
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Frame Rate</string>
             </property>
             <property name="buddy">
              <cstring>cmbFrameRate</cstring>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="cmbFrameRate">
             <property name="currentIndex">
              <number>-1</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Render Width</string>
             </property>
             <property name="buddy">
              <cstring>sbRenderWidth</cstring>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="sbRenderWidth">
             <property name="suffix">
              <string>px</string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>10000</number>
             </property>
             <property name="value">
              <number>1920</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Render Height</string>
             </property>
             <property name="buddy">
              <cstring>sbRenderHeight</cstring>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="sbRenderHeight">
             <property name="suffix">
              <string>px</string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>10000</number>
             </property>
             <property name="value">
              <number>1080</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Order</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QWidget" name="widget_2" native="true">
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <property name="spacing">
               <number>3</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QRadioButton" name="rbRowMajorLayout">
                <property name="text">
                 <string>Row-major</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">bgrpCellOrder</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="rbColMajorLayout">
                <property name="text">
                 <string>Column-major</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">bgrpCellOrder</string>
                </attribute>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Row Count</string>
             </property>
             <property name="buddy">
              <cstring>sbRowCount</cstring>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QSpinBox" name="sbRowCount">
             <property name="minimum">
              <number>1</number>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Column Count</string>
             </property>
             <property name="buddy">
              <cstring>sbColCount</cstring>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QSpinBox" name="sbColCount">
             <property name="minimum">
              <number>1</number>
             </property>
            </widget>
           </item>
           <item row="6" column="0" colspan="2">
            <widget class="Line" name="line_3">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="7" column="0" colspan="2">
            <widget class="QCheckBox" name="chbDebugVis">
             <property name="text">
              <string>Debug Visualization</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="btnStartRender">
      <property name="text">
       <string>Start</string>
      </property>
      <property name="default">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Preview</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="ResizableQLabel" name="lbCanvas">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar">
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="textVisible">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="btnStopRender">
      <property name="text">
       <string>Stop</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>16</width>
     <height>16</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionRender"/>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1117</width>
     <height>25</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::FolderOpen"/>
   </property>
   <property name="text">
    <string>Open MIDI</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRender">
   <property name="icon">
    <iconset theme="QIcon::ThemeIcon::CameraVideo"/>
   </property>
   <property name="text">
    <string>Render</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ResizableQLabel</class>
   <extends>QLabel</extends>
   <header>resizableqlabel.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>scrollArea</tabstop>
  <tabstop>cmbFrameRate</tabstop>
  <tabstop>sbRenderWidth</tabstop>
  <tabstop>sbRenderHeight</tabstop>
  <tabstop>rbRowMajorLayout</tabstop>
  <tabstop>rbColMajorLayout</tabstop>
  <tabstop>sbRowCount</tabstop>
  <tabstop>sbColCount</tabstop>
  <tabstop>chbDebugVis</tabstop>
  <tabstop>cmbChannel</tabstop>
  <tabstop>chbInheritOpts</tabstop>
  <tabstop>btnResetOpts</tabstop>
  <tabstop>scraChannelOpts</tabstop>
  <tabstop>lineEdit</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>dsbWaveThickness</tabstop>
  <tabstop>sbScopeWidth</tabstop>
  <tabstop>dsbAmplification</tabstop>
  <tabstop>chbStereo</tabstop>
  <tabstop>sbMaxNudge</tabstop>
  <tabstop>dsbTriggerThreshold</tabstop>
  <tabstop>btnStartRender</tabstop>
  <tabstop>btnStopRender</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>cmbChannel</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setCurrentChannel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>367</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>363</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbRenderWidth</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>recalcPreview()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>134</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbRenderHeight</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>recalcPreview()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>165</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bgrpCellOrder</sender>
   <signal>idClicked(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>recalcPreview()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbRowCount</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>recalcPreview()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbColCount</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>recalcPreview()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>278</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnStartRender</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>debugStart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>568</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>576</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnResetOpts</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>resetCurrentToDefault()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>130</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chbInheritOpts</sender>
   <signal>clicked(bool)</signal>
   <receiver>scraChannelOpts</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>287</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>456</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>debugStart()</slot>
  <slot>debugStop()</slot>
  <slot>setCurrentChannel(int)</slot>
  <slot>recalcPreview()</slot>
  <slot>resetCurrentToDefault()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="bgrpCellOrder"/>
 </buttongroups>
</ui>
